src/metatensor/models/utils/errors.py

```diff
@@ -0,0 +1,20 @@
+class ArchitectureError(Exception):
+    """
+    Exception raised for errors originating from architectures
+
+    This exception should be raised when an error occurs within an architecture's
+    operation, indicating that the problem is not directly related to the
+    metatensor-models infrastructure but rather to the specific architecture being used.
+
+    :param exception: The original exception that was caught, which led to raising this
+        custom exception.
+    :type exception: The exception message includes the message of the original
+        exception, followed by a note emphasizing that the error likely originates from
+        an architecture.
+    """
+
+    def __init__(self, exception):
+        super().__init__(
+            f"{exception}\n\nThis error originates from an architecture, and is likely "
+            "not a problem with metatensor-models."
```

Comment:
I feel this will make our users think they're dealing with a collection of models rather
than a piece of useful software (and therefore will not give us credit)
```suggestion
            f"{exception}\n\nThis error originates from an architecture. If you think this is "
            "a bug, please contact its maintainer (see the architecture's documentation)."
```

Reply:
Actually, "you can contact its maintainer" is even better IMO

Reply:
Yes this is a better. I will change it and also swap the order. First the we give the test and below state the error message.